### ISSUE TITLE
JavaDecaf cannot define an auxiliary class with the same name as the main script

### ISSUE THREAD

**Issue by sergutsan** [SophieKoonin/JavaDecaf#9]
JavaDecaf produces a cryptic error when compiling a script defines an internal class with the same name as the Java Decaf script. For example, a file called Point.jdc with the following code: 

```
Point myPoint;

class Point {
  int x;
  int y;
}
```

produces the following rather cryptic (for a novice) error: 

```
JavaDecaf: Compiling file Point.jdc...
Point.java:10: error: class Point is already defined in package unnamed package
    static class Point {
           ^ 
1 error
JavaDecaf: Compilation finished in 24 ms
```

There are two possibilities of fixing this: either an error is produced that says that JavaDecaf does not allow classes to have the same name as the script in which they are, or the name of internal classes is changed in some way before writing their definition to the Java source. I think the first one is simpler and is good enough.

**Comment by sophiekoonin** [comment]
This is causing problems I didn't anticipate - what should be a fairly simple fix (check the class name against the file name) is causing trouble when it checks whether it's a decaf nested class or not. The parser refuses to see the variable jjtThis (i.e. the tree). Will have another look soon.

**Comment by sergutsan** [comment]
I have managed to detect this in `ASTNestedClassDeclaration` by checking 

```
if (parser.getClassName().equals(begin.next.image))
```
- `parser.getClassName()` returns the name of the main JavaDecaf script
- `begin` is the token before the current one
- `begin.next` is the first of the current node´s tokens

Not sure if this is the best way of doing this. 0:-) Seems a bit hacky searching for the current token by looking at the next of the previous. %-/

I do not really know what to do once I detect that this is the case. Should I throw an exception? I did not see any other similar case in the classes under `ast/`. 

Maybe `ASTNestedClassDeclaration` is not even the right place to address this issue.
